src/wsdl.rs

```diff
@@ -227,83 +229,133 @@ impl<'a, 'input> WsPortOperation<'a, 'input> {
 
     /// Retrieve the input message for this port.
     pub fn input(&self) -> Result<Option<WsMessage<'a, 'input>>> {
-        let message_typename = match self
-            .0
-            .children()
-            .find(|n| n.has_tag_name(("http://schemas.xmlsoap.org/wsdl/", "input")))
-            .map(|n| n.attribute("message"))
-            .flatten()
-        {
-            Some(n) => n,
-            None => return Ok(None),
+        let mut inputs = self.inputs()?;
+        let Some(input) = inputs.next() else {
+            return Ok(None);
         };
+        if let Some((name, extra)) = inputs.next() {
+            return Err(WsError::new(
+                extra.node(),
+                WsErrorType::MalformedWsdl(WsErrorMalformedType::TooManyElements(format!(
+                    "extra `inputs` element of: {name}"
+                ))),
+            ));
+        }
+        Ok(Some(input.1))
+    }
 
-        let (_message_namespace, message_name) =
-            split_qualified(message_typename).map_err(|e| WsError::new(self.0, e))?;
-
+    /// Only a single input is allowed to exist so this is non-public.
+    fn inputs(&self) -> Result<impl Iterator<Item = (&'a str, WsMessage<'a, 'input>)>> {
         let def = WsDefinitions::find_parent(self.0)?;
-        Ok(Some(
-            def.messages()?
-                .find(|n| n.0.attribute("name") == Some(message_name))
-                .ok_or(WsError::new(
-                    self.0,
-                    WsErrorType::InvalidReference(message_name.to_string()),
-                ))?,
-        ))
+        Ok(self
+            .0
+            .children()
+            .filter(|n| n.has_tag_name(("http://schemas.xmlsoap.org/wsdl/", "input")))
+            .filter_map(|n| Some((n.attribute("name")?, n.attribute("message")?)))
+            .filter_map(
+                |(name, message_typename)| match split_qualified(message_typename) {
+                    Ok((_message_namespace, message_name)) => Some((name, message_name)),
+                    Err(_) => None,
+                },
+            )
+            .filter_map(move |(name, message_name)| {
+                let Ok(mut messages) = def.messages() else {
+                    return None;
+                };
+                if let Some(message) =
+                    messages.find(|n| n.0.attribute("name") == Some(message_name))
+                {
+                    Some((name, message))
+                } else {
+                    None
+                }
+            }))
     }
```

Comment:
Ditto - build a list of `inputs` and then error out if there is more than one :)